pkg/clients/cloudantdatabase/cloudantdatabase.go

```diff
@@ -0,0 +1,129 @@
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
+
+	cv1 "github.com/IBM/cloudant-go-sdk/cloudantv1"
+
+	"github.com/crossplane-contrib/provider-ibm-cloud/apis/cloudantv1/v1alpha1"
+	ibmc "github.com/crossplane-contrib/provider-ibm-cloud/pkg/clients"
+)
+
+// LateInitializeSpec fills optional and unassigned fields with the values in *cv1.Databaseinformation object.
+func LateInitializeSpec(spec *v1alpha1.CloudantDatabaseParameters, in *cv1.DatabaseInformation) error { // nolint:gocyclo
+
+	if spec.Partitioned == nil {
+		spec.Partitioned = in.Props.Partitioned
+	}
+
+	if spec.Q == nil {
+		spec.Q = in.Cluster.Q
+	}
+
+	return nil
+}
+
+// GenerateCreateCloudantDatabaseOptions produces PutDatabaseOptions object from CloudantDatabaseParameters object.
+func GenerateCreateCloudantDatabaseOptions(in v1alpha1.CloudantDatabaseParameters, o *cv1.PutDatabaseOptions) error {
+	o.Db = reference.ToPtrValue(in.Db)
+	o.Partitioned = in.Partitioned
+	o.Q = in.Q
+	return nil
+}
+
+// GenerateObservation produces CloudantDatabaseObservation object from *cv1.DatabaseInformation object.
+func GenerateObservation(in *cv1.DatabaseInformation) (v1alpha1.CloudantDatabaseObservation, error) {
+	o := v1alpha1.CloudantDatabaseObservation{
+		Cluster:            Generatev1alpha1DatabaseInformationCluster(in.Cluster),
+		CommittedUpdateSeq: reference.FromPtrValue(in.CommittedUpdateSeq),
+		CompactRunning:     ibmc.BoolValue(in.CompactRunning),
+		CompactedSeq:       reference.FromPtrValue(in.CompactedSeq),
+		DiskFormatVersion:  ibmc.Int64Value(in.DiskFormatVersion),
+		DocCount:           ibmc.Int64Value(in.DocCount),
+		DocDelCount:        ibmc.Int64Value(in.DocDelCount),
+		Engine:             reference.FromPtrValue(in.Engine),
+		Sizes:              Generatev1alpha1ContentInformationSizes(in.Sizes),
+		UpdateSeq:          reference.FromPtrValue(in.UpdateSeq),
+		UUID:               reference.FromPtrValue(in.UUID),
+	}
+	return o, nil
+}
+
+// Generatev1alpha1DatabaseInformationCluster generates *v1alpha1.DatabaseinformationCluster from *cv1.DatabaseInformationCluster
+func Generatev1alpha1DatabaseInformationCluster(in *cv1.DatabaseInformationCluster) *v1alpha1.DatabaseInformationCluster {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.DatabaseInformationCluster{
+		N: ibmc.Int64Value(in.N),
+		R: ibmc.Int64Value(in.R),
+		W: ibmc.Int64Value(in.W),
+	}
+	return o
+}
+
+// Generatev1alpha1ContentInformationSizes generates *v1alpha1.ContentInformationSizes from *cv1.ContentInformationSizes
+func Generatev1alpha1ContentInformationSizes(in *cv1.ContentInformationSizes) *v1alpha1.ContentInformationSizes {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.ContentInformationSizes{
+		Active:   ibmc.Int64Value(in.Active),
+		External: ibmc.Int64Value(in.External),
+		File:     ibmc.Int64Value(in.File),
+	}
+	return o
+}
+
+// how should I ensure isuptodate always returns true, is this good enough ?? 
```

Comment:
Since this needs to always return true you can remove all the logic (lines 102-116) and just have `return true, nil`. Just add a comment to document that there is no update that can be done for the Cloudant DB API so this method has to always return true